engines/director/lingo/xlibs/mystisle.cpp

```diff
@@ -0,0 +1,110 @@
+#include "director/lingo/lingo-utils.h"
+#include "director/lingo/xlibs/mystisle.h"
+
+/**************************************************
+ *
+ * USED IN:
+ * Mysterious Island: A Race Against Time and Hot Lava
+ *
+ **************************************************/
+
+/*
+-- MystIsle External Factory, v. 1.0 by David Jackson-Shields
+--MYSTISLE
+I      mNew                     --Creates new instance
+X      mDispose                 --Disposes of XObject instance
+S      mName                    --Returns the XObject name (MystIsle)
+I      mGetLastError            --Returns most recent error code
+I      mGetSysVersion           --returns MS Windows version as an integer
+I      mGetQTVersion            --returns QuickTime version as an integer
+I      mVerifyUnlocked          --verify startup volume is writable (for Prefs)
+S      mGetPrefsPath            --return path of Windows directory
+IPP    mDiagnostic, picHdl1, picHdl2 --for testing during development
+ */
+
+namespace Director {
+
+const char *MystIsleXObj::xlibName = "MystIsle";
+const XlibFileDesc MystIsleXObj::fileNames[] = {
+	{ "MYSTISLE",   nullptr },
+	{ nullptr,        nullptr },
+};
+
+static MethodProto xlibMethods[] = {
+	{ "new",				MystIsleXObj::m_new,		 0, 0,	400 },
+	{ "dispose",				MystIsleXObj::m_dispose,		 0, 0,	400 },
+	{ "name",				MystIsleXObj::m_name,		 0, 0,	400 },
+	{ "getLastError",				MystIsleXObj::m_getLastError,		 0, 0,	400 },
+	{ "getSysVersion",				MystIsleXObj::m_getSysVersion,		 0, 0,	400 },
+	{ "getQTVersion",				MystIsleXObj::m_getQTVersion,		 0, 0,	400 },
+	{ "verifyUnlocked",				MystIsleXObj::m_verifyUnlocked,		 0, 0,	400 },
+	{ "getPrefsPath",				MystIsleXObj::m_getPrefsPath,		 0, 0,	400 },
+	{ "diagnostic",				MystIsleXObj::m_diagnostic,		 2, 2,	400 },
+	{ nullptr, nullptr, 0, 0, 0 }
+};
+
+static BuiltinProto xlibBuiltins[] = {
+
+	{ nullptr, nullptr, 0, 0, 0, VOIDSYM }
+};
+
+MystIsleXObject::MystIsleXObject(ObjectType ObjectType) :Object<MystIsleXObject>("MystIsle") {
+	_objType = ObjectType;
+}
+
+void MystIsleXObj::open(ObjectType type, const Common::Path &path) {
+    MystIsleXObject::initMethods(xlibMethods);
+    MystIsleXObject *xobj = new MystIsleXObject(type);
+    if (type == kXtraObj)
+        g_lingo->_openXtras.push_back(xlibName);
+    g_lingo->exposeXObject(xlibName, xobj);
+    g_lingo->initBuiltIns(xlibBuiltins);
+}
+
+void MystIsleXObj::close(ObjectType type) {
+    MystIsleXObject::cleanupMethods();
+    g_lingo->_globalvars[xlibName] = Datum();
+
+}
+
+void MystIsleXObj::m_new(int nargs) {
+	g_lingo->printSTUBWithArglist("MystIsleXObj::m_new", nargs);
+	g_lingo->dropStack(nargs);
+	g_lingo->push(g_lingo->_state->me);
+}
+
+XOBJSTUBNR(MystIsleXObj::m_dispose)
+XOBJSTUB(MystIsleXObj::m_name, "MystIsle")
+XOBJSTUB(MystIsleXObj::m_getLastError, 0)
+XOBJSTUB(MystIsleXObj::m_getSysVersion, 700)  // game checks < 310 (Win) or < 700 (Mac)
+XOBJSTUB(MystIsleXObj::m_getQTVersion, 200)  // game checks < 110 (Win) or < 200 (Mac)
```

Comment:
Would it be better to unstub this and implement a check based on machine type is mac or windows?

See also this code for context: https://github.com/scummvm/scummvm/blob/master/engines/director/lingo/lingo-the.cpp#L542

Reply:
Done